tests/monitors/df/df_test.py

```diff
@@ -1,67 +1,105 @@
 
 pytestmark = [pytest.mark.collectd, pytest.mark.df, pytest.mark.monitor_without_endpoints]
 
 METADATA = Metadata.from_package("collectd/df")
 
 
-def test_df():
-    with Agent.run(
+def test_df_included_metrics():
+    expected_metrics = METADATA.included_metrics
+    verify_custom(
         """
         monitors:
           - type: collectd/df
             hostFSPath: /
-        """
-    ) as agent:
-        _ = (
-            wait_for(lambda: set(agent.fake_services.datapoints_by_metric) == METADATA.included_metrics),
-            "timed out waiting for metrics and/or dimensions!",
-        )
-        assert set(agent.fake_services.datapoints_by_metric) == METADATA.included_metrics
-        assert not has_log_message(agent.output.lower(), "error"), "error found in agent output!"
+        """,
+        expected_metrics,
+    )
 
 
 def test_df_extra_metrics():
-    expected_metrics = METADATA.included_metrics | {"df_complex.reserved"}
-    with Agent.run(
-        """
+    extra_metric_1, extra_metric_2 = "df_complex.reserved", "df_inodes.reserved"
+    expected_metrics = METADATA.included_metrics | {extra_metric_1, extra_metric_2}
+    verify_custom(
+        f"""
         monitors:
           - type: collectd/df
             hostFSPath: /
             extraMetrics:
-            - df_complex.reserved
-        """
-    ) as agent:
-        _ = (
-            wait_for(lambda: set(agent.fake_services.datapoints_by_metric) == expected_metrics),
-            "timed out waiting for metrics and/or dimensions!",
-        )
-        assert set(agent.fake_services.datapoints_by_metric) == expected_metrics
-        assert not has_log_message(agent.output.lower(), "error"), "error found in agent output!"
+            - {extra_metric_1}
+            - {extra_metric_2}
+        """,
+        expected_metrics,
+    )
+
+
+def test_df_percent_prefixed_metrics():
+    expected_metrics = METADATA.included_metrics | {
+        "percent_bytes.free",
+        "percent_bytes.reserved",
+        "percent_bytes.used",
+        "percent_inodes.free",
+        "percent_inodes.reserved",
+        "percent_inodes.used",
+    }
+    percent_prefixed_wildcard = "percent_*"
+    verify_custom(
+        f"""
+        monitors:
+          - type: collectd/df
+            hostFSPath: /
+            extraMetrics:
+            - {percent_prefixed_wildcard}
+        """,
+        expected_metrics,
+    )
+
+
+def test_df_invalid_extra_metric():
+    expected_metrics = METADATA.included_metrics
+    invalid_extra_metric = "Y2W8OBrdZZ"
+    verify_custom(
+        f"""
+        monitors:
+          - type: collectd/df
+            hostFSPath: /
+            extraMetrics:
+            - {invalid_extra_metric}
+        """,
+        expected_metrics,
+    )
+
+
+# TODO: Fix failing test
+@pytest.mark.skip(reason="failing due to bug")
+def test_df_blank_extra_metric():
+    expected_metrics = METADATA.included_metrics
+    blank_extra_metric = " "
```

Comment:
Doesn't need to be a test case for df monitor.

Reply:
The monitor is supposed to behave as for invalid extra metric. In that case a warning is issued and included metrics are sent according to @keitwb. But no metrics are sent in this case.